packages/cli/src/lib/emitUsage.ts

```diff
@@ -0,0 +1,77 @@
+import { Usage } from '@appland/client';
+import { Metadata } from '@appland/models';
+import { Git } from '../telemetry';
+import type { UsageReport, UsageUpdateDto } from '@appland/client/dist/src/usage';
+import sanitizeURL from './repositoryInfo';
+
+function serializeMetadata(metadata?: Metadata): string | undefined {
+  if (!metadata) return;
```

Comment:
How could this be falsey?

Reply:
As it's implemented now it should never be undefined. This was more of a code style decision to satisfy the type checker when returning an array element. E.g.

As implemented I can:
```ts
serializeMetadata(metadataArray[0]);
```
vs. without:
```ts
if (metadataArray.length) serializeMetadata(metadataArray[0]);
```

After collecting and adding [git information to the metadata](https://github.com/getappmap/appmap-js/pull/1302/commits/64b4e4439f464c7065e987ff4668c274b8c19f89#diff-19be78c4bb538809c1cbfc6fa61ff897e0db6f959d0ec2bd791675fb740706c8R8-R25) this will now always return a value.